uaclient/api/u/pro/status/notices/v1.py

```diff
@@ -0,0 +1,105 @@
+import logging
+from typing import List
+
+from uaclient.api.api import APIEndpoint
+from uaclient.api.data_types import AdditionalInfo
+from uaclient.config import UAConfig
+from uaclient.data_types import DataObject, Field, StringDataValue, data_list
+from uaclient.files.notices import Notice, NoticesManager
+
+LOG = logging.getLogger("ubuntupro.lib.auto_attach")
+
+
+class NoticeInfo(DataObject):
+    fields = [
+        Field("order_id", StringDataValue, doc="Notice order id"),
+        Field(
+            "message",
+            StringDataValue,
+            doc="Message displayed by the notice",
+        ),
+        Field("label", StringDataValue, doc="Notice label"),
+    ]
+
+    def __init__(self, order_id: str, message: str, label: str):
+        self.order_id = order_id
+        self.message = message
+        self.label = label
+
+
+class NoticeListResult(DataObject, AdditionalInfo):
+    fields = [
+        Field(
+            "notices",
+            data_list(NoticeInfo),
+            doc="A list of currently saved notices",
+        ),
+    ]
+
+    def __init__(self, notices: List[NoticeInfo]):
+        self.notices = notices
+
+
+def notice_list() -> NoticeListResult:
+    return _get_notice_list(cfg=UAConfig())
+
+
+def _get_notice_list(cfg: UAConfig) -> NoticeListResult:
+    _notice_manager = NoticesManager()
+    noticeList = _notice_manager.get_active_notices()
+    notices = []
+    for notice in noticeList:
+        for n in Notice:
```

Comment:
Why iterate through Notice?

Reply:
Updated and added a unit test for better coverage @orndorffgrant